homeassistant/components/nut/__init__.py

```diff
@@ -107,6 +121,35 @@ async def _async_update_listener(hass: HomeAssistant, entry: ConfigEntry):
     await hass.config_entries.async_reload(entry.entry_id)
 
 
+def _unique_id_from_nut_server(config):
+    """Get the mac address of NUT if its on the LAN."""
+    nut_host = config[CONF_HOST]
+    host_is_ip = True
+
+    if nut_host == LOCALHOST:
+        nut_host = IPV4_LOOPBACK
+    else:
+        try:
+            ipaddress.ip_address(nut_host)
+        except ValueError:
+            host_is_ip = False
+
+    if host_is_ip and is_loopback(ipaddress.ip_address(nut_host)):
+        mac_addr = getmac.get_mac_address()
+    elif host_is_ip:
+        mac_addr = getmac.get_mac_address(ip=nut_host)
+    else:
+        mac_addr = getmac.get_mac_address(hostname=nut_host)
+
+    _LOGGER.debug("mac_address of NUT at %s: %s", nut_host, mac_addr)
+
+    if mac_addr is None or mac_addr == EMPTY_MAC_ADDRESS:
+        return None
+
+    ups_alias = config.get(CONF_ALIAS, DEFAULT_UPS_ALIAS)
```

Comment:
We shouldn't use aliases in unqiue IDs, they are easily changed. Why not do mac + port ?

Reply:
They can help multiple upses connected.  If they change the alias they have to resetup the config entry.

Reply:
Switching to using the config entry ID per out of band conversion